### ISSUE TITLE
Performing "The Ultimate CF" test. Result : Flare trashing Spinal Modem makes Spinal Modem not do brain damage.

### ISSUE THREAD

**Issue by PesiTheWizard** [mtgred/netrunner#2619]
See title.
Chat log follows.
This scenario started, with fewer cards involved, as a judge-ruling stating that "when trace successful" triggers all happen before the "if successful" result of the trace; though I can't seem to find it again.
If you see any other bugs here, please make note of them.

> valranoth spends to play Mad Dash.
Pesi
Initiating trace...
Pesi uses Flare to initiate a trace with strength 6 (6 + 0 ) (Trash 1 hardware, do 2 meat damage, and end the run).
valranoth
Boosting link by 0
valranoth spends 0 to increase link strength to 0.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
Pesi
Note: Runner credits at 1.
Pesi
Clicking hardware to trash now....
Pesi uses Flare to trash Spinal Modem.
valranoth uses Muresh Bodysuit to prevent the first meat damage this turn.
valranoth places 1 power counter on Clan Vengeance.
valranoth places 1 power counter on Clan Vengeance.
valranoth trashes I've Had Worse, I've Had Worse due to damage.
valranoth uses I've Had Worse to draw 3 cards.
valranoth uses I've Had Worse to draw 3 cards.
valranoth uses Synthetic Blood to draw 1 card.
Pesi uses Aryabhata Tech to gain 1 and force the Runner to lose 1 .
valranoth uses Order of Sol to gain 1 .
valranoth trashes Dadiana Chacon and suffers 3 meat damage.
valranoth places 1 power counter on Clan Vengeance.
valranoth places 1 power counter on Clan Vengeance.
valranoth trashes Careful Planning, Careful Planning, Code Siphon due to damage.
Pesi uses ChiLo City Grid to give the Runner 1 tag.
valranoth uses Jesminder Sareen: Girl Behind the Curtain to avoid the first tag during this run.
Pesi trashes Autoscripter.
valranoth uses John Masanori to take 1 tag.
valranoth trashes Rachel Beckman for being tagged.
Pesi uses Mr. Stone to do 1 meat damage.
valranoth places 1 power counter on Clan Vengeance.
valranoth places 1 power counter on Clan Vengeance.
valranoth trashes Data Breach due to damage.
Pesi uses Georgia Emelyov to do 1 net damage.
valranoth places 1 power counter on Clan Vengeance.
valranoth places 1 power counter on Clan Vengeance.
valranoth
Waiting for Net Analytics...
Pesi
Net Analytics pops up
Pesi
Clicking "yes"
Pesi uses Net Analytics to draw a card.
valranoth
Waiting for Tori Hanzo
Pesi
Tori Hanzo pops up
Pesi
Clicking yes
Pesi uses Tori Hanzō to do 1 brain damage instead of net damage.
Pesi uses Defective Brainchips to do 1 additional brain damage.
valranoth places 1 power counter on Clan Vengeance.
valranoth places 1 power counter on Clan Vengeance.
Pesi uses Sentinel Defense Program to do 1 net damage.
valranoth places 1 power counter on Clan Vengeance.
valranoth places 1 power counter on Clan Vengeance.
valranoth trashes Cold Read due to damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
valranoth places 1 power counter on Clan Vengeance.
valranoth places 1 power counter on Clan Vengeance.
valranoth trashes I've Had Worse due to damage.
valranoth uses I've Had Worse to draw 3 cards.
valranoth trashes Cold Read, Careful Planning due to damage.
valranoth suffers 1 meat damage from Mad Dash.
valranoth places 1 power counter on Clan Vengeance.
valranoth places 1 power counter on Clan Vengeance.
valranoth trashes Code Siphon due to damage.
Pesi
Run over

The reason for the double Clan Vengeance triggers is that there were two of them installed, before and after installing all the rest, just in case it mattered. Same with the scoring of Sentinel Defense Program.
The corp ID was Seidr Labs

**Comment by PesiTheWizard** [comment]
Other issues noticed :
2 brain damage was done, yet the Sentinel Defense Programs only did 2. Either one of them failed entirely or they only noticed the first instance of brain damage.

Clan Vengeance gains counters for the net damage that Tori Hanzo turned into brain damage. What if the brain damage were to be prevented?

**Comment by JoelCFC25** [comment]
This is really hard to follow. It would be much more valuable to have smaller bug demonstrations involving only the afflicted cards with a discrete (hopefully small) set of reproduction steps.

**Comment by JoelCFC25** [comment]
OK, the Spinal Modem bit is easy at any rate--it's simply missing `:delayed-completion true`.

**Comment by JoelCFC25** [comment]
Turns out that this Flare + Spinal Modem bug is not easy after all. Still stumped.

**Comment by JoelCFC25** [comment]
I have this one now. Any Corp ability for successful traces is actually firing before the `:successful-trace` event in `resolve-trace`.

**Comment by PesiTheWizard** [comment]
Friday 15. sept 2017 post-update
1 Sentinel Defense Program, Clan Vengeance missing, Defective Brainchips missing.

> LynxMegaCorp spends to play Mad Dash.
Pesi
Choosing Trace strength...
Pesi uses Flare to initiate a trace with strength 6 (6 + 0 ) (Trash 1 hardware, do 2 meat damage, and end the run).
LynxMegaCorp spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.   <- _Wut._
Pesi uses Sentinel Defense Program to do 1 net damage.
Pesi uses Aryabhata Tech to gain 1 and force the Runner to lose 1 .
LynxMegaCorp uses Order of Sol to gain 1 .
LynxMegaCorp trashes Dadiana Chacon and suffers 3 meat damage.
LynxMegaCorp uses Muresh Bodysuit to prevent the first meat damage this turn.
Pesi uses ChiLo City Grid to give the Runner 1 tag.
LynxMegaCorp uses Jesminder Sareen: Girl Behind the Curtain to avoid the first tag during this run.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
Pesi
Tori Hanzo trigger. Choosing yes.
Pesi uses Tori Hanzō to do 1 brain damage instead of net damage.   <- _SDP triggers on the wut_
Pesi uses Sentinel Defense Program to do 1 net damage.
LynxMegaCorp trashes I've Had Worse due to damage.   <- _1st trash to damage_
LynxMegaCorp uses I've Had Worse to draw 3 cards.
LynxMegaCorp uses Synthetic Blood to draw 1 card.
LynxMegaCorp trashes Autoscripter due to damage.   <- _2nd trash to damage_
Pesi
Net Analytics trigger. Choosing Yes
Pesi uses Net Analytics to draw a card.
Pesi
Choosing hardware to trash with Flare
Pesi uses Flare to trash Spinal Modem.
LynxMegaCorp trashes I've Had Worse, Spinal Modem due to damage.   <- _3rd trash to damage (2)_
LynxMegaCorp uses I've Had Worse to draw 3 cards.
LynxMegaCorp suffers 1 meat damage from Mad Dash.
LynxMegaCorp trashes Levy AR Lab Access due to damage.   <- _4th trash to damage_
LynxMegaCorp uses John Masanori to take 1 tag.
LynxMegaCorp trashes Rachel Beckman for being tagged.
Pesi uses Mr. Stone to do 1 meat damage.
LynxMegaCorp trashes Mad Dash due to damage.   <- _5th trash to damage_
Pesi trashes Autoscripter.
Pesi uses Georgia Emelyov to do 1 net damage.
LynxMegaCorp trashes Peace in Our Time due to damage.   <- _6th trash to damage_

Other than the corp somehow taking damage, it seems like not enough damage was done.
Maybe the "corp takes damage" thing is only a log fail, in which case Muresh Bodysuit probably prevented ALL the damage from Dadiana Chacon; or the runner didn't take the damage assigned to the corp, in which case I have no idea what's going on.

**Comment by JoelCFC25** [comment]
I'm sorry, but I just cannot untangle all of that to understand the core issue. Is there anyway to isolate a limited set of cards that are causing....whatever the issue is?

**Comment by JoelCFC25** [comment]
> Pesi suffers 1 brain damage from Spinal Modem.   <- Wut.

This at least is cosmetic only, an easily corrected usage of `system-msg`.

**Comment by PesiTheWizard** [comment]
I did this with all the cards to see if there was actually something wrong somewhere. Now that we know where to look, we can narrow things down.

**Comment by PesiTheWizard** [comment]
Tests performed 3rd/Feb/2018

Flare+Spinal Only:

> Redino987 spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.   <-- _Name thing still happening_
Redino987 trashes Imp due to damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
Pesi uses Flare to trash Spinal Modem.
Redino987 trashes Imp, Imp due to damage.

Flare+Spinal and Defective Brainchips:

> Redino987 spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Defective Brainchips to do 1 additional brain damage.
Redino987 trashes Imp, Imp due to damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
Pesi uses Flare to trash Spinal Modem.
Redino987 trashes Imp, Imp due to damage.

Flare+Spinal and 1*SDP:

> Redino987 spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Sentinel Defense Program to do 1 net damage.
Redino987 trashes Imp due to damage.
Redino987 trashes Imp due to damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
Pesi uses Flare to trash Spinal Modem.
Redino987 trashes Imp, Imp due to damage.

Flare+Spinal, Defective Brainchips and 1*SDP:

> Redino987 spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Defective Brainchips to do 1 additional brain damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Redino987 trashes Imp due to damage.
Redino987 trashes Imp, Imp due to damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
Pesi uses Flare to trash Spinal Modem.
Redino987 trashes Imp, Imp due to damage.

The above + Gerogia Emelyov:

> Redino987 spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Defective Brainchips to do 1 additional brain damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Redino987 trashes Imp due to damage.
Redino987 trashes Imp, Imp due to damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
Pesi uses Flare to trash Spinal Modem.
Redino987 trashes Imp, Imp due to damage.
Pesi uses Georgia Emelyov to do 1 net damage.
Redino987 trashes Imp due to damage.

The above + Tori Hanzo:

> Redino987 spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Defective Brainchips to do 1 additional brain damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
_Tori asks_
Pesi uses Tori Hanzo to do 1 brain damage instead of net damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Redino987 trashes Imp due to damage.
Redino987 trashes Imp due to damage.
Pesi uses Flare to trash Spinal Modem.
Pesi uses Georgia Emelyov to do 1 net damage.

The above + ChiLo City Grid+Mr. Stone:

> Redino987 spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Defective Brainchips to do 1 additional brain damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Pesi uses ChiLo City Grid to give the Runner 1 tag.
Pesi uses Mr. Stone to do 1 meat damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
_Tori asks_
Pesi uses Tori Hanzo to do 1 brain damage instead of net damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Redino987 trashes Imp due to damage.
Redino987 trashes Imp due to damage.
Pesi uses Flare to trash Spinal Modem.
Pesi uses Georgia Emelyov to do 1 net damage.

The above + a 2nd SDP:

> Redino987 spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Defective Brainchips to do 1 additional brain damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Pesi uses ChiLo City Grid to give the Runner 1 tag.
Pesi uses Mr. Stone to do 1 meat damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
_Tori asks_
Pesi uses Tori Hanzo to do 1 brain damage instead of net damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Redino987 trashes Imp due to damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Redino987 trashes Imp due to damage.
Redino987 trashes Imp due to damage.
Pesi uses Flare to trash Spinal Modem.
Pesi uses Georgia Emelyov to do 1 net damage.

**Comment by PesiTheWizard** [comment]
Tests performed 11/Mar/2018

Flare+Spinal, 1*SDP, Defective Brainchips,Georgia Emelyov,ChiLo City Grid+Mr. Stone,Aryabhata Tech

> Billrye spends to make a run on Server 2.
Pesi uses Flare to initiate a trace with strength 6 (6 + 0 ) (Trash 1 hardware, do 2 meat damage, and end the run).
Billrye spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Defective Brainchips to do 1 additional brain damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Billrye trashes Earthrise Hotel due to damage.
Billrye trashes Maw, Maw due to damage.
Pesi uses Aryabhata Tech to gain 1 and force the Runner to lose 1 .
Pesi uses ChiLo City Grid to give the Runner 1 tag.
Pesi uses Mr. Stone to do 1 meat damage.
Billrye trashes Earthrise Hotel due to damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
Pesi uses Flare to trash Spinal Modem.
Billrye trashes Daily Casts, Sure Gamble due to damage.
Pesi uses Georgia Emelyov to do 1 net damage.
Billrye trashes Daily Casts due to damage.

Flare+Spinal, 2*SDP, Defective Brainchips,Georgia Emelyov,ChiLo City Grid+Mr. Stone,Aryabhata Tech

> Billrye spends to make a run on Server 2.
Pesi uses Flare to initiate a trace with strength 6 (6 + 0 ) (Trash 1 hardware, do 2 meat damage, and end the run).
Billrye spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Defective Brainchips to do 1 additional brain damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Billrye trashes Career Fair due to damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Billrye trashes Levy AR Lab Access due to damage.
Billrye trashes Sure Gamble, Earthrise Hotel due to damage.
Pesi uses Aryabhata Tech to gain 1 and force the Runner to lose 1 .
Pesi uses ChiLo City Grid to give the Runner 1 tag.
Pesi uses Mr. Stone to do 1 meat damage.
Billrye trashes Hacktivist Meeting due to damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
Pesi uses Flare to trash Spinal Modem.
Billrye trashes Liberated Account, Liberated Account due to damage.
Pesi uses Georgia Emelyov to do 1 net damage.
Billrye trashes Hacktivist Meeting due to damage.

The above + Tori(saying no)

> Billrye spends to make a run on Server 2.
Pesi uses Flare to initiate a trace with strength 6 (6 + 0 ) (Trash 1 hardware, do 2 meat damage, and end the run).
Billrye spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Defective Brainchips to do 1 additional brain damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Pesi uses Aryabhata Tech to gain 1 and force the Runner to lose 1 .
Pesi uses ChiLo City Grid to give the Runner 1 tag.
Pesi uses Mr. Stone to do 1 meat damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
_Tori popup(no)_
Billrye trashes Acacia due to damage.
Pesi uses Flare to trash Spinal Modem.
Billrye trashes Black Orchestra, Yusuf due to damage.
Pesi uses Georgia Emelyov to do 1 net damage.
Billrye trashes MKUltra due to damage.

Same, but saying yes to Tori

> Billrye spends to make a run on Server 2.
Pesi uses Flare to initiate a trace with strength 6 (6 + 0 ) (Trash 1 hardware, do 2 meat damage, and end the run).
Billrye spends 0 to increase link strength to 0.
Pesi suffers 1 brain damage from Spinal Modem.
Pesi uses Sentinel Defense Program to do 1 net damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Pesi uses Aryabhata Tech to gain 1 and force the Runner to lose 1 .
Pesi uses ChiLo City Grid to give the Runner 1 tag.
Pesi uses Mr. Stone to do 1 meat damage.
Pesi uses Flare to trash 1 hardware, do 2 meat damage, and end the run.
_Tori popup(yes)_
Pesi uses Tori Hanzō to do 1 brain damage instead of net damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Billrye trashes Virus Breeding Ground due to damage.
Pesi uses Sentinel Defense Program to do 1 net damage.
Billrye trashes Sure Gamble due to damage.
Billrye trashes Yusuf due to damage.
Pesi uses Flare to trash Spinal Modem.
Pesi uses Georgia Emelyov to do 1 net damage.

**Comment by JoelCFC25** [comment]
So are all of these wrong results? I am not following. They would be immensely more useful if there was a clear summary of the correct result vs. what the actual result.

**Comment by PesiTheWizard** [comment]
Let's call the 3rd/Feb/2018 tests "Set 1" and the 11/Mar/2018 tests "Set 2"
What I have so far:

Set 1 test 4 shows that instead of Defective Brainchips doing its own 1 brain damage, it is pulling a The Cleaners and increasing the brain damage done by another source as seen by the fact that the Sentinel Defense Program does only 1 net damage total, and that the brain damages are both in a single log entry. This may be how the card is supposed to work, I haven't looked at the FAQs lately.

I should note that after Tori is added in Set 1, always I said yes to the prompt.

The difference between Set 1 test 5 and Set 1 test 6 shows the effects that Tori's presence has. In test 5, 6 cards were trashed, in test 6 only 2. All the damage that has stacked up until Tori's popup shows up is 'on hold' somehow, and when she does the brain damage - followed by another net damage - those seem to be the only damages that are suffered, the rest somehow being forgotten.
In accordance with that new Tori bug report, Emelyov logs damage but no card is trashed.

I did not know when I performed Set 1 that Tori would be this overwhelming. I had planned to add cards until something broke, and Tori basically broke all subsequent tests.
This is also the point at which I wished that damage-trashing logs would specify the type of damage.
So for Set 2 I decided to save her inclusion until last.

Set 2 tests 1 and 2: Everything is handled beautifully. No complaints (provided Defective Brainchips is working as intended, see first paragraph.)

Set 2 test 3 shows the "forgotten damage" thing more starkly, that it also happens if saying no, and that saying no does not result in the "no subsequent net damage"-thing reported earlier today.

Set 2 test 4 was an experimental failure.
